### DISCUSSION TITLE
Clearing the application cache causes the cache invalidator to fail

### DISCUSSION THREAD

**Discussion by miicah** [statamic/cms#8502] [Troubleshooting]
Versions: can replicate on 3.4 and 4.8
Static caching strategy: full

Steps to reproduce:

1. Visit a page so a static page is generated
2. Edit the content on that page and click publish
   - The file in the static cache is correctly invalidated and deleted
   - Visit the page again and the static page is created
3. Now go and clear the application cache
4. Go back to the page, change some content and click publish
5. The page is not invalidated and thus the original static file remains in the static folder

From what I can understand, this is happening because clearing the application cache means the entire stache is blown away and there is never a call to go back and re-index (re-check?) all of the files in the content folders.

**Why do we do this?** Because we need to upload very large files to Statamic and cannot use the CMS, so they are copied directly into the assets folder - unfortunately this won't show up in the CMS until the application cache is cleared. This is how we came across the issue, but there are many reasons you might need to clear the application cache.

The simple solution is to run `php please static:warm` but this doesn't work for us as our production server is behind authentication, so the guzzle requests don't actually hit any of the real pages.

What I'm trying to understand is if this is a bug or an oversight in how full static caching is implemented? Should a `php please stache:warm` scan the static folders as well?

**Comment by ryanmitchell** [comment]
You may know this already but you can pass authentication params to the static warm command.

<img width="549" alt="Screenshot 2023-07-28 at 07 44 48" src="https://github.com/statamic/cms/assets/51899/fc3d4613-ac22-4902-9b09-3f806aada4cb">

Rather than clearing your application cache when you upload images did you try just running stache:clear? Or even just running stache:warm?

**Reply by miicah** [reply]
Thanks for the reply.

I'll address the auth question first, we can't pass credentials like that as our auth page is a portal (think like a Google OATH token) so the guzzle requests just hit the portal page and return a 200 response like everything is fine and the page has been warmed. Unless I'm misunderstanding your answer, I don't think we are able to get around this?

We recently did another upload (dragging files directly into the assets folder over FTP) and I tried:

- Using the CMS, click the 'Warm' button
- Using the CLI, `php please stache:warm`
- Using the CMS, click the 'Clear' button
- Using the CLI, `php please stache:clear` then `php please stache:warm`

None of these made our recently uploaded ZIP files show up in the assets collection.

On the CMS page for the stache it states: "It is generated automatically from the content files"

Where does the stache 'read' from to determine what things should be included? Does it look in the application cache or does it actually scan the folders directly?
